### DISCUSSION TITLE
I found a key function named "LockNode" when i review kwok code, what the lock means?

### DISCUSSION THREAD

**Discussion by sologgfun** [kubernetes-sigs/kwok#283] [Q&A] [answered]
I even review the first commit of kwok , I found the following code.
In my opinion, there are nothing about lock ,so i confused about the function name.
Did I miss something? If any refrence can give will be great , thank you!

```
// LockNode locks a given node
func (c *NodeController) LockNode(ctx context.Context, nodeName string) error {
  node, err := c.clientSet.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
  if err != nil {
    return err
  }

  patch, err := c.configureNode(node)
  if err != nil {
    return err
  }
  if patch == nil {
    return nil
  }
  node, err = c.clientSet.CoreV1().Nodes().PatchStatus(ctx, node.Name, patch)
  if err != nil {
    return err
  }
  c.logger.Printf("Lock node %s", nodeName)
  return nil
}
```

**Comment by wzshiming** [comment, marked as answer]
https://github.com/wzshiming/fake-kubelet/blob/a6c1082175f151ae643e2d2e63b00035bb315bed/controller.go#L135

Derived from Kwok's predecessor, [fake-kubelet](https://github.com/wzshiming/fake-kubelet), the first version of which meant locking the resource in the ready status, so that no one could modify it

**Reply by sologgfun** [reply]
ok, i get it,thanks 😄
